lucene/core/src/java/org/apache/lucene/util/bkd/OfflinePointWriter.java

```diff
@@ -54,7 +54,10 @@ public void append(byte[] packedValue, int docID) throws IOException {
     assert closed == false : "Point writer is already closed";
     assert packedValue.length == config.packedBytesLength : "[packedValue] must have length [" + config.packedBytesLength + "] but was [" + packedValue.length + "]";
     out.writeBytes(packedValue, 0, packedValue.length);
-    out.writeInt(docID);
+    out.writeByte((byte) (docID >> 24));
+    out.writeByte((byte) (docID >> 16));
+    out.writeByte((byte) (docID >> 8));
+    out.writeByte((byte) (docID >> 0));
```

Comment:
oh tricky, is this because we need the serialized order to compare lexicographically? If so, can you leave a comment? And maybe do a `writeInt(Integer.reverseBytes(docID))` to keep it more readable?

Reply:
done